Detectors/ITSMFT/ITS/workflow/src/DCSAdaposParserSpec.cxx

```diff
@@ -0,0 +1,219 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   DCSAdaposParserSpec.cxx
+
+#include "ITSWorkflow/DCSAdaposParserSpec.h"
+
+namespace o2
+{
+namespace its
+{
+
+//////////////////////////////////////////////////////////////////////////////
+// Default constructor
+ITSDCSAdaposParser::ITSDCSAdaposParser()
+{
+  this->mSelfName = o2::utils::Str::concat_string(ChipMappingITS::getName(), "ITSDCSAdaposParser");
+}
+
+//////////////////////////////////////////////////////////////////////////////
+void ITSDCSAdaposParser::init(InitContext& ic)
+{
+  LOGF(info, "ITSDCSAdaposParser init...", mSelfName);
+
+  this->mCcdbUrl = ic.options().get<std::string>("ccdb-url");
+
+  this->mVerboseOutput = ic.options().get<bool>("verbose");
+
+  // Read alpide param object from ccdb: this is the first read, object will be refreshed in run()
+  this->mCcdbFetchUrl = ic.options().get<std::string>("ccdb-fetch-url");
+  getCurrentCcdbAlpideParam();
+
+  return;
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// Main running function
+// Get Data from ADAPOS and prepare CCDB objects
+void ITSDCSAdaposParser::run(ProcessingContext& pc)
+{
+
+  // Retrieve adapos data and process them
+  auto dps = pc.inputs().get<gsl::span<DPCOM>>("input");
+  process(dps);
+
+  if (doStrobeUpload && dps.size() > 0) {
+    // upload to ccdb
+    pushToCCDB(pc);
+    // refresh the local alpide param object
+    getCurrentCcdbAlpideParam();
```

Comment:
Now you are querying only after update, when your alpideParam is anyway supposed to be in sync with CCDB value. The only point to query would be if CCDB updates from other source are possible (or if the CCDB upload of this workflow is not done to the source CCDB). But, in this case you should recheck not after the update by in some time intervals, e.g. if > 10 s elapsed from the previous query.

Reply:
ok now I got the full point. I was thinking that I had to call `get` every time I wanted to update the local ccdb object. Now I did the right thing based on your suggestion.